Makefile

```diff
@@ -1,23 +1,35 @@
+# Whether to turn compiler warnings into errors
+export WERROR ?= true
+
 default: release
 
 release:
-	mkdir -p build && cd build && cmake ../ -DCMAKE_BUILD_TYPE=Release && VERBOSE=1 cmake --build .
+	mkdir -p build/Release && cd build/Release && cmake ../../ -DCMAKE_BUILD_TYPE=Release -DWERROR=$(WERROR) && VERBOSE=1 cmake --build .
```

Comment:
@springmeyer I'm noticing this is much different than the bench and the hpp-skel master Makefiles. Curious if you have a preference. I also noticed that this conflicts with how clang-tidy runs (I have it setup locally, not in this branch). Perhaps it has something to do with the extra `../`?

Reply:
>  I'm noticing this is much different than the bench and the hpp-skel master Makefiles. 

This was done so we could have `make test` run both release and debug tests while developing. I think we should roll back to match `hpp-skel` now that development is slowing down.

>  I also noticed that this conflicts with how clang-tidy runs (I have it setup locally, not in this branch).

Whoops, that was an oversight. So, hopefully rolling back to match `hpp-skel` will help the clang-tidy scripts find things right.

Reply:
Syncing with master and I'll keep the WERROR glad in there